File: DesignPatternsExercises/java/ProxyDesignPattern/Exercise/MovieProxy.java
class MovieProxy __________(2)__________ VideoFunctions {
    private Movie movie;

    public MovieProxy(String name) {
        movie = __________(3)__________(name);
    }

    @Override
    public void playMovie() {
        movie.__________(4)__________();
    }

    @Override
    public void stopMovie() {
        movie.__________(5)__________();
    }

    @Override
    public void __________(6)__________() {
        System.out.println("Deleting movie: Permission denied!!!");
    }

    @Override
    public String getMovieDetails() {
        return movie.__________(7)__________();
    }
}

File: DesignPatternsExercises/ruby/proxy_design_pattern/solution/movie_proxy.rb
require_relative 'movie'
require_relative 'video_functions'

class MovieProxy < VideoFunctions
  def initialize(name)
    # @movie = __________(2)__________
    @movie = Movie.new(name)
  end

  def play_movie
    # __________(3)__________
    @movie.play_movie
  end

  def stop_movie
    # __________(4)__________
    @movie.stop_movie
  end

  def delete_movie
    puts 'Deleting movie: Permission denied!!!'
  end

  def movie_details
    # __________(5)__________
    @movie.movie_details
  end
end
